helpers/hasCorrectNextConfig.js

```diff
@@ -12,9 +11,29 @@ const hasCorrectNextConfig = async ({ nextConfigPath, failBuild }) => {
   const isValidTarget = acceptableTargets.includes(target)
   if (!isValidTarget) {
     console.log(
-      `Your next.config.js must set the "target" property to one of: ${acceptableTargets.join(', ')}. Update the
-      target property to allow this plugin to run.`,
+      `The "target" config property must be one of "${acceptableTargets.join(
+        '", "',
+      )}". Building with "serverless" target.`,
     )
+
+    /* eslint-disable fp/no-delete, node/no-unpublished-require */
+
+    // We emulate Vercel so that we can set target to serverless if needed
+    process.env.NOW_BUILDER = true
+    // If no valid target is set, we use an internal Next env var to force it
+    process.env.NEXT_PRIVATE_TARGET = 'serverless'
+
+    // 🐉 We need Next to recalculate "isZeitNow" var so we can set the target, but it's
+    // set as an import side effect so we need to clear the require cache first. 🐲
+    // https://github.com/vercel/next.js/blob/canary/packages/next/telemetry/ci-info.ts
+
+    delete require.cache[require.resolve('next/dist/telemetry/ci-info')]
+    delete require.cache[require.resolve('next/dist/next-server/server/config')]
```

Comment:
Considering `next` is performing some logic at `require()`-time, this approach makes sense.

The only alternative I can think of would be to call `getNextConfig()` in a separate process. However, this would add some significant complexity. Also, this would JSON-serialize the Next.js configuration, which would not work if the config includes non-JSON values (can it?), unless we use `serialization: 'advanced'` but that option is not available in Node 10.

Two potential problems we might have in the future though are:
  - I am unsure whether this would work with ES modules since the current Node.js implementation does not allow removing cached URLs (it allows passing query parameters as cache key, but this would not work here), so this might require mixing CommonJS `require()` and ES modules `import` in order to keep using `require.cache`.
  - This relies on `next` internal file structure. The cache deletion would be a noop if they choose to rename those files. Some things we could do to fix this:
     - Run some automated tests confirming those files exist with the latest version of `next`
     - `delete` all the `next` entries in `require.cache` instead of those specific files